### ISSUE TITLE
[Feature] Add SELinux support

### ISSUE THREAD

**Issue by LFRon** [waydroid/waydroid#1179]
### Is your feature request related to a problem? Please describe.

SELinux is very important to Android.Many apps couldn't run on non-SELinux Android device.
And Magisk drop the support of non-SELinux device on Magisk 26.4-Canary.It means waydroid couldn't install magisk newer version.I hope Waydroid could support this as quickly as possible😭.

### Describe the solution you'd like.

Maybe it could use qemu+kvm or crosvm to run waydroid.Its performance very good.I tried on i3-4005U(@1.6Ghz,even worse than Snapdragon 660),the android running on crosvm is very smooth and it fully support SELinux.

### Describe alternatives you've considered.

Or just let LXC could use SELinux.I remebered Waydroid supports SELinux to be permissive when I use Ubuntu 22.04 last year.

**Comment by Quackdoc** [comment]
no, qemu or crosvm is not an option. It may be possible to properly use SElinux with waydroid, not too sure, but considering that waydroid already supports overlays like magisk does, im not sure anyone is too interested in working on it atm

**Comment by ChrysoliteAzalea** [comment]
Waydroid is a container (a purely user-space application that runs Android in an isolated environment), so in order to use SELinux, it has to be enabled on the host. Then, you need SELinux policies for Waydroid -- as far as I can remember, Fedora packaged their own, but I'm not sure, I'm not using it. The problem may be that the container software not interacting with SELinux (it was patched in order to be usable on non-SELinux systems since requiring SELinux would make it incompatible with AppArmor, btw, it's the same with AppArmor -- Android software doesn't know about AppArmor, and all domain transitions are handled by the kernel based on profiles only).

> Maybe it could use qemu+kvm or crosvm to run waydroid.

Waydroid is a container solution, switching to QEMU would lose the benefits of a container and increase overhead -- meanwhile, you can already run Android in a QEMU virtual machine (using BlissOS, for example).

**Comment by LFRon** [comment]
> Waydroid is a container (a purely user-space application that runs Android in an isolated environment), so in order to use SELinux, it has to be enabled on the host. Then, you need SELinux policies for Waydroid -- as far as I can remember, Fedora packaged their own, but I'm not sure, I'm not using it. The problem may be that the container software not interacting with SELinux (it was patched in order to be usable on non-SELinux systems since requiring SELinux would make it incompatible with AppArmor, btw, it's the same with AppArmor -- Android software doesn't know about AppArmor, and all domain transitions are handled by the kernel based on profiles only).
> 
> > Maybe it could use qemu+kvm or crosvm to run waydroid.
> 
> Waydroid is a container solution, switching to QEMU would lose the benefits of a container and increase overhead -- meanwhile, you can already run Android in a QEMU virtual machine (using BlissOS, for example).

I tried crosvm android on ChromeOS on my junk pc (i3-4005U @1.6Ghz).Crosvm running it smoothly.And I found that ChromeOS use the host kernel with kvm to run android inside
![Screenshot_2023-11-30-21-36-48-174_com qihoo contents](https://github.com/waydroid/waydroid/assets/141307807/0d8c36c7-ca8d-401b-85f5-04f55fd743b2)

**Comment by oponic** [comment]
That's what a container does.

**Comment by LFRon** [comment]
> no, qemu or crosvm is not an option. It may be possible to properly use SElinux with waydroid, not too sure, but considering that waydroid already supports overlays like magisk does, im not sure anyone is too interested in working on it atm

Linux distros are use AppArmor more than SELinux,I have searched on the Internet and it says it should open SELinux on the host.I'd like to use VM because I think gfxstream is a good choice😂

**Comment by Quackdoc** [comment]
Then use a virtual machine. BlissOS supports Virgl which works on both CrosVM and QEMU. Some people have thought about getting GFXstream working on it. No one as far as I know has started working on it. However, with Virtio Venus coming it's not super necessary.

**Comment by LFRon** [comment]
> Then use a virtual machine. BlissOS supports Virgl which works on both CrosVM and QEMU. Some people have thought about getting GFXstream working on it. No one as far as I know has started working on it. However, with Virtio Venus coming it's not super necessary.

I'd like gfxstream because it could realize the wayland-passthrough on qemu.Crosvm for Linux has too many problems😂.Wayland-passthrough means it makes it possible to realize the waydroid multi_window function
